### ISSUE TITLE
Cannot load JSON file

### ISSUE THREAD

**Issue by saturngod** [zed-industries/zed#9639]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Open [usaCities.json](https://gist.github.com/saturngod/629e62a7fee28a8246104956c6621a24)
2. Zed became white screen and all the text has been lost

### Environment

Mac OS 14.3.1
Apple M1 Pro

### If applicable, add mockups / screenshots to help explain present your vision of the feature

Tested on Zed

<img width="2032" alt="Screenshot 2024-03-21 at 9 45 34 AM" src="https://github.com/zed-industries/zed/assets/135177/d9478f74-47d2-43c5-b758-c5869a714780">

Tested on VSCode

<img width="1470" alt="Screenshot 2024-03-21 at 9 45 21 AM" src="https://github.com/zed-industries/zed/assets/135177/fe0922ef-525f-409a-84e2-12c0fb1e2ac0">

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

[Zed.log](https://github.com/zed-industries/zed/files/14696879/Zed.log)

**Comment by Moshyfawn** [comment]
I can't seem to reproduce with [this gist](https://gist.github.com/ahmu83/38865147cf3727d221941a2ef8c22a77) in the latest [`v0.128.1`](https://zed.dev/releases/preview/0.128.1) preview with any theme, as your linked file doesn't seem to exist.

**Comment by saturngod** [comment]
@Moshyfawn  , I created the json file on https://gist.github.com/saturngod/629e62a7fee28a8246104956c6621a24 . This file doesn't support the correct JSON format. It's using single quote instead of double quote. With the correct JSON format, zed doesn't have the issue.

**Comment by Moshyfawn** [comment]
Interestingly, the Zed log is spammed with

```log
[ERROR] scene too large: 0 paths, 3 shadows, 23329 quads, 64 underlines, 1293 mono, 2 poly, 0 surfaces
```

as well as the grey editor area is actually rendered on top of any piece of UI, so trying to open the command palette works, but you can't see the UI element.

**Comment by notpeter** [comment]
I attempted to reproduce this with the json file linked and Zed displays the file correctly, renders 23908 diagnostic errors ("json: Value expected") without any log spam.

That said, it does hang if you try and open the diagnostics panel. So I've opened a distinct issue for that here: 

Thanks for reporting and for the pathologically bad JSON file @saturngod. This is definitely giving Zed a workout.
